foodbornenyc/sources/twitter_search.py

```diff
@@ -154,8 +161,10 @@ def query_twitter(how_long=0, interval=5):
             continue
         #logger.info("%i Total unique tweets in %i:%i:%i time", len(id_set),
         #            *sec_to_hms(time()-start))
-        new_tweets = [ t for t in tweets if t['id_str'] not in id_set ]
-        new_Tweets = tweets_to_Tweets(new_tweets, fields)
+        new_tweets = [t for t in tweets if t['id_str'] not in id_set]
+        # if a retrieved tweet has a loc/user with a matching ID already in the
+        # db, that loc/user is updated instead of a new one added, bc of merge
+        new_Tweets = [db.merge(t) for t in tweets_to_Tweets(new_tweets, fields)]
```

Comment:
I just tested without the id set (commented line 70 that filled it in) and made Twitter return the same tweet json both times. It successfully adds a single tweet to the DB, so I think `db.merge` works as expected.

I'll figure out how to do a speed test to compare the two solutions.